src/devices/wholeBodyDynamics/WholeBodyDynamicsDevice.cpp

```diff
@@ -508,6 +522,34 @@ bool WholeBodyDynamicsDevice::openExternalWrenchesPorts(os::Searchable& config)
     return ok;
 }
 
+bool WholeBodyDynamicsDevice::openFilteredFTPorts(os::Searchable& config)
+{
+    bool ok = true;
+        // create port names
+
+        std::string sensorName;
+        std::string portName;
+        outputFTPorts.resize(estimator.sensors().getNrOfSensors(iDynTree::SIX_AXIS_FORCE_TORQUE));
+        for(int ft=0; ft < estimator.sensors().getNrOfSensors(iDynTree::SIX_AXIS_FORCE_TORQUE); ft++ )
+        {
+            sensorName= estimator.sensors().getSensor(iDynTree::SIX_AXIS_FORCE_TORQUE,ft)->getName() ;
+
+            yDebug() << "wholeBodyDynamics: creating port names for  " << sensorName;
+
+            portName=(portPrefix+"/filteredFT/"+sensorName);
+            //std::unique_ptr<yarp::os::BufferedPort<yarp::sig::Vector>> outputPort
+            outputFTPorts[ft]=std::make_unique<yarp::os::BufferedPort<yarp::sig::Vector> >();
```

Comment:
Is there any reason why you are using a vector of pointers to ports instead of just a vector of ports?

Reply:
I got the idea from this https://stackoverflow.com/questions/6624819/c-vector-of-objects-vs-vector-of-pointers-to-objects and the way the current external ports are opened.
My idea is that maybe later instead of just a Boolean I can decide which ft ports to open. I'll see if it's worth given the usage of the feature. So the array may change size. Although it might be Overkill in here since the other external ports are in fact I side a structure while this ones are not.